### PR TITLE
[Messenger] Removed deprecated code

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

This remove deprecated code from Messenger. 

### TODO 
- [x] Write changelog

### CODE REVIEW

**Review by derrabus — CHANGES_REQUESTED** [review]
Can you have a look at the Psalm errors regarding `AbstractFailedMessagesCommand`?

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 1]
```diff
@@ -34,15 +34,11 @@ class MessengerPass implements CompilerPassInterface
...
-        if (0 < \func_num_args()) {
-            trigger_deprecation('symfony/messenger', '5.3', 'Configuring "%s" is deprecated.', __CLASS__);
-        }
-
-        $this->handlerTag = $handlerTag;
-        $this->busTag = $busTag;
-        $this->receiverTag = $receiverTag;
+        $this->handlerTag = 'messenger.message_handler';
+        $this->busTag = 'messenger.bus';
+        $this->receiverTag = 'messenger.receiver';
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: We should inline the strings and remove the constructor.

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 2]
```diff
@@ -30,20 +30,9 @@
...
-            trigger_deprecation('symfony/messenger', '5.3', 'Configuring "%s" is deprecated.', __CLASS__);
-        }
-
-        $this->handlerTag = $handlerTag;
-        $this->busTag = $busTag;
-        $this->receiverTag = $receiverTag;
-    }
+    private $handlerTag = 'messenger.message_handler';
+    private $busTag = 'messenger.bus';
+    private $receiverTag = 'messenger.receiver';
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: Let's inline the properties as well. https://github.com/symfony/symfony/pull/40468#issuecomment-845791510

Comment (Nyholm) [reply src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: Sure

**Comment by Nyholm** [comment]
The PR is updated and rebased.

**Comment by Nyholm** [comment]
Status: needs review

File: src/Symfony/Component/Messenger/Bridge/Amqp/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+6.0
+---
+
+ * Remove option `prefetch_count`.
+ * Using invalid options will throw an `LogicException`.
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/Bridge/Amqp/CHANGELOG.md]:
```suggestion
 * Using invalid options will throw a `LogicException`.
```

File: src/Symfony/Component/Messenger/Bridge/Redis/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+6.0
+---
+
+ * Remove option `tls`.
+ * Using invalid options will throw an `LogicException`.
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/Bridge/Redis/CHANGELOG.md]:
```suggestion
 * Using invalid options will throw a `LogicException`.
```

File: src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php [hunk 5]
```diff
@@ -46,7 +46,7 @@ abstract class AbstractFailedMessagesCommand extends Command
     /**
      * @param ServiceProviderInterface $failureTransports
      */
-    public function __construct(?string $globalFailureReceiverName, $failureTransports)
+    public function __construct(?string $globalFailureReceiverName, ?ServiceProviderInterface $failureTransports)
     {
         $this->failureTransports = $failureTransports;
         if (!$failureTransports instanceof ServiceProviderInterface) {
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php]: Can't we remove this BC layer? Same for the `getReceiverName()` method below?

Comment (Nyholm) [reply src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php]: But you are not always required to pass a `ServiceProviderInterface`, right? 

Are you still allowed to do null? 

I started working on removing this, but I didn't quite understand what the end result should look like. I either need some guidance or I'll leave this to another PR (which Im currently doing)

File: src/Symfony/Component/Messenger/Stamp/RedeliveryStamp.php [hunk 6]
```diff
@@ -21,23 +20,11 @@ final class RedeliveryStamp implements StampInterface
 {
     private $retryCount;
     private $redeliveredAt;
-    private $exceptionMessage;
-    private $flattenException;
 
-    public function __construct(int $retryCount, string $exceptionMessage = null, FlattenException $flattenException = null, \DateTimeInterface $redeliveredAt = null)
+    public function __construct(int $retryCount, \DateTimeInterface $redeliveredAt = null)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Stamp/RedeliveryStamp.php:24]: It looks like the BC layer is broken in 5.2: passing $redeliveredAt as the second argument should be allow already.
Can you please submit a PR doing so?

File: src/Symfony/Component/Messenger/Bridge/Amqp/CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+6.0
+---
+
+ * Remove option `prefetch_count`.
+ * Using invalid options will throw a `LogicException`.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Bridge/Amqp/CHANGELOG.md]: no final dots please :)

File: src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php (hunk 5 above)
Comment (chalasr) [reply src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php]: This is handled in #41878

**Comment by nicolas-grekas** [comment]
Rebase needed!

File: src/Symfony/Component/Messenger/Stamp/RedeliveryStamp.php (hunk 6 above)
Comment (Nyholm) [reply src/Symfony/Component/Messenger/Stamp/RedeliveryStamp.php:24]: Sure. #42059

**Review by Nyholm — COMMENTED** [review]
PR is updated and rebased. 

Status: needs review

File: src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php [hunk 8]
```diff
@@ -56,7 +55,7 @@ class Connection
     private $deleteAfterReject;
     private $couldHavePendingMessages = true;
 
-    public function __construct(array $configuration, array $connectionCredentials = [], array $redisOptions = [], \Redis|\RedisCluster $redis = null)
+    public function __construct(array $configuration, array $connectionCredentials = [], array $redisOptions = [], \Redis | \RedisCluster $redis = null)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php]: should be reverted (same elsewhere)

Comment (Nyholm) [reply src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php]: Okey. 
FYI: This came form fabbot

Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php]: yep, I'm looking forward to https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5734

**Comment by nicolas-grekas** [comment]
Thank you @Nyholm.

**Comment by Nyholm** [comment]
Thank you for merging

**Pull request merged**